bolt/lib/Core/BinarySection.cpp

```diff
@@ -190,18 +190,7 @@ void BinarySection::flushPendingRelocations(raw_pwrite_stream &OS,
   clearList(PendingRelocations);
 }
 
-BinarySection::~BinarySection() {
-  if (isReordered()) {
```

Comment:
Why are we dropping this condition?

Reply:
Now that we manage the memory inside `updateContents()` it's not needed and can lead to double-free.